core/src/main/java/org/elasticsearch/search/sort/FieldSortBuilder.java

```diff
@@ -66,25 +100,28 @@ public FieldSortBuilder order(SortOrder order) {
         return this;
     }
 
+    /** Returns the order of sorting. */
+    public SortOrder order() {
+        return this.order;
+    }
+
     /**
      * Sets the value when a field is missing in a doc. Can also be set to <tt>_last</tt> or
      * <tt>_first</tt> to sort missing last or first respectively.
      */
     @Override
     public FieldSortBuilder missing(Object missing) {
-        this.missing = missing;
+        if (missing instanceof String) {
+            this.missing = BytesRefs.toBytesRef(missing);
+        } else {
+            this.missing = missing;
+        }
         return this;
     }
 
-    /**
-     * Sets if the field does not exists in the index, it should be ignored and not sorted by or not. Defaults
-     * to <tt>false</tt> (not ignoring).
-     * @deprecated Use {@link #unmappedType(String)} instead.
-     */
-    @Deprecated
-    public FieldSortBuilder ignoreUnmapped(boolean ignoreUnmapped) {
-        this.ignoreUnmapped = ignoreUnmapped;
-        return this;
+    /** Returns the value used when a field is missing in a doc. */
+    public Object missing() {
+        return this.missing;
```

Comment:
In BaseTermQueryBuilder we convert BytesRef back to String in the getter, we could do here as well, otherwise client setting a String gets something different back here.